autoloads/save_manager.gd

```diff
@@ -0,0 +1,174 @@
+extends Node
+
+#
+# Variables
+#
+
+## This is where the save file is stored
+var save_file_path = "user://save_game.data"
+
+## These store the current checkpoint and collected presents of the player
+var current_checkpoint: Checkpoint = null
+var current_saved_collection: Dictionary = {}
+
+## This is what is used to spawn the player at a specific position when loading
+var load_position: Vector2
+
+## These bools handle the save text fading up and down and also whether or not 
+## we're ready for saving
+var ready_to_save: bool = false
+var show_saved_text = false
+var get_rid_of_saved_text = false
+
+## This is when we start fading up or down
+var start_time: float = -1
+
+## This is the time it takes for the save text to fade up and a third of the
+## time it takes for it to fade down
+var time_text_fade = 1500
+
+## This is a reference to the actual text to display
+var saved_text: Label
+
+func _ready():
+	# Connect up the signals needed
+	CheckpointManager.checkpoint_flagged.connect(on_checkpoint_flagged.bind())
+	PresentManager.collected_updated.connect(on_collection_saved.bind())
+	get_node("/root/GameEntry").level_loaded.connect(on_scene_ready_for_load.bind())
+
+func _process(delta):
+	# Set start time to an actual value to use for math
+	if start_time == -1:
+		start_time = Time.get_ticks_msec()
+	
+	if show_saved_text:
+		# Fade text up
+		saved_text.add_theme_color_override("font_color", lerp(Color(1, 1, 1, 0), Color(1, 1, 1, 1)\
+		, (Time.get_ticks_msec() - start_time) / time_text_fade))
+
+		if Time.get_ticks_msec() - start_time >= time_text_fade:
+			# Set up the fade down and stop us from fading up again
+			start_time = -1
+			show_saved_text = false
+			get_rid_of_saved_text = true
+			saved_text.add_theme_color_override("font_color", Color(1, 1, 1, 1))
+	
+	elif saved_text != null and get_rid_of_saved_text:
+		# Make the fade back 3 times longer to let the user have a chance to read it
+		# Fade text down
+		saved_text.add_theme_color_override("font_color", lerp(Color(1, 1, 1, 1), Color(1, 1, 1, 0)\
+		, (Time.get_ticks_msec() - start_time) / time_text_fade / 3))
+
+		if Time.get_ticks_msec() - start_time >= time_text_fade * 3:
+			# Make sure we don't fade down again
+			get_rid_of_saved_text = false
+			saved_text.add_theme_color_override("font_color", Color(1, 1, 1, 0))
+
+func save_game():
+	# If we have a checkpoint to save to
+	if current_checkpoint != null:
```

Comment:
IMO you should use a guard clause here, because this whole function is in the if statement!

Reply:
Fair enough, that would probably make it better on the cache, wasn't really thinking about that when writing this.